src/data/equiv/encodable/basic.lean

```diff
@@ -2,31 +2,42 @@
 Copyright (c) 2015 Microsoft Corporation. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Leonardo de Moura, Mario Carneiro
-
-Type class for encodable Types.
-Note that every encodable Type is countable.
 -/
 import data.equiv.nat
 import order.rel_iso
 import order.directed
 
+/-!
+# Encodable types
+
+This file defines encodable (constructively countable) types as a typeclass extending `encodable`.
+This is used to provide explicit encode/decode functions from and to `ℕ`, with the information that
+those functions are inverses of each other.
+
+## Main declarations
+
+
+
+## Implementation notes
+
+The point of asking for an explicit partial inverse `decode : ℕ → option α` to `encode : α → ℕ` is
+to make the range of `encode` decidable even when the finiteness of `α` is not.
+-/
+
 open option list nat function
 
-/-- An encodable type is a "constructively countable" type. This is where
-  we have an explicit injection `encode : α → nat` and a partial inverse
-  `decode : nat → option α`. This makes the range of `encode` decidable,
-  although it is not decidable if `α` is finite or not. -/
+/-- Constructively countable type. Made from an explicit injection `encode : α → ℕ` and a partial
```

Comment:
There is a subtlety in this definition, that it works for finite types (including the empty type) and infinite countable types. This is worth mentioning it here and in the file-level docstring. You could also mention here the `denumerable` typeclass` (which only works for infinite countable types).